src/App.tsx

```diff
@@ -1,26 +1,60 @@
-// https://reactjs.org/docs/code-splitting.html
+/*
+  https://reactjs.org/docs/code-splitting.html
+  https://reactrouter.com/en/main/route/loader
+  loader: function to provide data to the route element before it renders.
+  type vs interface: https://www.typescriptlang.org/play#example/types-vs-interfaces
+*/
+
 import { Suspense, lazy } from 'react';
 import { BrowserRouter as Router, Routes, Route } from "react-router-dom";
 import Layout from './components/Layout';
 import Spinner from "./components/Spinner";
-const Home = lazy(() => import('./pages/home/Home'));
-const About = lazy(() => import('./pages/About'));
-const Contact = lazy(() => import('./pages/Contact'));
-const Card = lazy(() => import('./pages/card/Card'));
+import PrivateRoute from './routes/PrivateRoute';
+import PublicRoute from './routes/PublicRoute';
+import Public from './routes/Public';
+import Private from './routes/Private';
+import _ from 'lodash';
```

Comment:
Just a small array, no need to use lib like lodash. If we had to use it import like this `import { map } from 'lodash'`